src/ketos/compile.rs

```diff
@@ -488,8 +488,6 @@ impl<'a> Compiler<'a> {
                         if self.load_local_name(name)? {
                             self.push_instruction(Instruction::Push)?;
                             pushed_fn = true;
-                        } else if self.self_name == Some(name) {
```

Comment:
Removing this branch changes the effect of this code.

If clippy is suggesting this change, I believe it is a bug.